test/jasmine/tests/click_test.js

```diff
@@ -817,6 +928,588 @@ describe('Test click interactions:', function() {
+            expect(pt.cyFinal).toEqual(160, 'points[0].cyFinal');
+            expect(pt.hidden).toEqual(false, 'points[0].hidden');
+            expect(pt.i).toEqual(4, 'points[0].i');
+            expect(pt.label).toEqual('4', 'points[0].label');
+            expect(pt.largeArc).toEqual(0, 'points[0].largeArc');
+            expect(pt.midangle).toEqual(1.0471975511965976, 'points[0].midangle');
+            expect(pt.px0).toEqual([0, -60], 'points[0].px0');
+            expect(pt.px1).toEqual([51.96152422706632, 29.999999999999986], 'points[0].px1');
+            expect(pt.pxmid).toEqual([51.96152422706631, -30.000000000000007], 'points[0].pxmid');
+            expect(pt.r).toEqual(60, 'points[0].r');
+            expect(typeof pt.t).toEqual(typeof {}, 'points[0].t');
+            expect(pt.text).toEqual('33.3%', 'points[0].text');
+            expect(typeof pt.trace).toEqual(typeof {}, 'points[0].trace');
+            expect(pt.v).toEqual(5, 'points[0].v');
+            expect(pt.vTotal).toEqual(15, 'points[0].vTotal');
+
+            var evt = futureData.event;
+            expect(evt.clientX).toEqual(pointPos[0], 'event.clientX');
+            expect(evt.clientY).toEqual(pointPos[1], 'event.clientY');
+            Object.getOwnPropertyNames(clickOpts).forEach(function(opt) {
+                expect(evt[opt]).toEqual(clickOpts[opt], 'event.' + opt);
+            });
+        });
+    });
+
+    describe('hover events', function() {
+        var futureData;
+
+        beforeEach(function(done) {
+            Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(done);
+
+            gd.on('plotly_hover', function(data) {
+                futureData = data;
+            });
+        });
+
+        it('should contain the correct fields', function() {
+            mouseEvent('mouseover', pointPos[0], pointPos[1]);
+
+            var point0 = futureData.points[0],
+                evt = futureData.event;
+            expect(point0).toEqual(evt, 'points');
+            expect(evt.clientX).toEqual(pointPos[0], 'event.clientX');
+            expect(evt.clientY).toEqual(pointPos[1], 'event.clientY');
+        });
+    });
+
+    describe('unhover events', function() {
+        var futureData;
+
+        beforeEach(function(done) {
+            Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(done);
+
+            gd.on('plotly_unhover', function(data) {
+                futureData = data;
+            });
+        });
+
+        it('should contain the correct fields', function() {
+            mouseEvent('mouseout', pointPos[0], pointPos[1]);
+
+            var point0 = futureData.points[0],
+                evt = futureData.event;
+            expect(point0).toEqual(evt, 'points');
+            expect(evt.clientX).toEqual(pointPos[0], 'event.clientX');
+            expect(evt.clientY).toEqual(pointPos[1], 'event.clientY');
+        });
+    });
+});
+
+
+describe('Test click interactions on a ternary plot:', function() {
```

Comment:
Similarly, move to https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/ternary_test.js